src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1550,6 +1569,10 @@ private function addDefaultParametersMethod(): string
 
     public function getParameter(string $name): array|bool|string|int|float|\UnitEnum|null
     {
+        if (isset(self::DEPRECATED_PARAMETERS[$name])) {
```

Comment:
@nicolas-grekas this change is a regression regarding @stof's comment that I had addressed.
It is not consistent anymore with the parameter bag logic, where we first check for a parameter existence before triggering a deprecation (https://github.com/symfony/symfony/pull/47719#discussion_r983595581).

Reply:
It makes me think about removing parameters, when it happens we should remove any relation deprecation as well to prevent such case in the dumper where we would trigger a deprecation before throwing a `ParameterNotFoundException`.
WDYT?

Reply:
indeed, we should only list params that exist in this array

Reply:
OK, I've pushed https://github.com/symfony/symfony/pull/47719/commits/81e63a734e787760f62e1c90c26a85207eaec791.